SDWebImage/SDWebImageManager.m

```diff
@@ -201,6 +205,10 @@ - (void)diskImageExistsForURL:(nullable NSURL *)url
                     }
                     
                     BOOL cacheOnDisk = !(options & SDWebImageCacheMemoryOnly);
+                    
+                    if (self.cacheKeyFilter && downloadedImage) {
```

Comment:
We need this extra scale because user may using cache filter to override your URLs. Since the scale process is really cheap, so we do not need to pass this `cacheKeyFilter` or `SDWebImageManager` instance  to that `SDWebImageDownloaderOperation`. And this can also make our class dependency clear